src/main/java/com/mycompany/mszczepienia/service/VisitService.java

```diff
@@ -112,6 +115,21 @@ public void updateMissedVisitsStatus() {
         });
     }
 
+    @Transactional
+    public VisitDto cancelVisit(Long visitId){
+        System.out.println(visitId);
+        Visit visit = visitRepository.getById(visitId);
+        System.out.println(visit);
+        visit.setVisitStatus(VisitStatus.CANCELLED);
+        visitRepository.saveAndFlush(visit);
+        return modelMapper.map(visit, VisitDto.class);
```

Comment:
There's also no need to return anything (especially when you use getById instead of findById. getBy... is a reference proxy and modelmapper call getters method internally for mapping.